### ISSUE TITLE
The number of images in trainset

### ISSUE THREAD

**Issue by TerryMelody** [GrandpaXun242/AdaBLDM#5]
Dear authors! Thanks for your contributions! When running python generate_dataset.py, you set GENERATENUM = 20000 to generate target images of trainset. It seems a large number for only a defect type of a class. What's the influence of different image numbers of trainset, for example, if I set GENERATENUM = 200?

**Comment by szgy66** [comment]
> Dear authors! Thanks for your contributions! When running python generate_dataset.py, you set GENERATENUM = 20000 to generate target images of trainset. It seems a large number for only a defect type of a class. What's the influence of different image numbers of trainset, for example, if I set GENERATENUM = 200?

I also have this doubt, MvtecAD data set has a total of 15 classes, each class contains several small classes, each small class claims that 20000 images is too time-consuming and the storage space is large, if I only train one class can, have you tried it?

**Comment by GrandpaXun242** [comment]
> Dear authors! Thanks for your contributions! When running python generate_dataset.py, you set GENERATENUM = 20000 to generate target images of trainset. It seems a large number for only a defect type of a class. What's the influence of different image numbers of trainset, for example, if I set GENERATENUM = 200?

You can just generate  the trainset of one class in one category , e.g. class "hazelnut-hole".
If  it set GENERATENUM = 200 , AdaBLDM maybe  not effective.

**Comment by GrandpaXun242** [comment]
> > Dear authors! Thanks for your contributions! When running python generate_dataset.py, you set GENERATENUM = 20000 to generate target images of trainset. It seems a large number for only a defect type of a class. What's the influence of different image numbers of trainset, for example, if I set GENERATENUM = 200?
> 
> I also have this doubt, MvtecAD data set has a total of 15 classes, each class contains several small classes, each small class claims that 20000 images is too time-consuming and the storage space is large, if I only train one class can, have you tried it?


Due to time constraints ， I only train 7 class of mvtec.

**Comment by TerryMelody** [comment]
> > > Dear authors! Thanks for your contributions! When running python generate_dataset.py, you set GENERATENUM = 20000 to generate target images of trainset. It seems a large number for only a defect type of a class. What's the influence of different image numbers of trainset, for example, if I set GENERATENUM = 200?
> > 
> > 
> > I also have this doubt, MvtecAD data set has a total of 15 classes, each class contains several small classes, each small class claims that 20000 images is too time-consuming and the storage space is large, if I only train one class can, have you tried it?
> 
> Due to time constraints ， I only train 7 class of mvtec.

I got it. Thanks for your reply. Is there a least number of generated images to make this model effective？

**Comment by GrandpaXun242** [comment]
> > > > Dear authors! Thanks for your contributions! When running python generate_dataset.py, you set GENERATENUM = 20000 to generate target images of trainset. It seems a large number for only a defect type of a class. What's the influence of different image numbers of trainset, for example, if I set GENERATENUM = 200?
> > > 
> > > 
> > > I also have this doubt, MvtecAD data set has a total of 15 classes, each class contains several small classes, each small class claims that 20000 images is too time-consuming and the storage space is large, if I only train one class can, have you tried it?
> > 
> > 
> > Due to time constraints ， I only train 7 class of mvtec.
> 
> I got it. Thanks for your reply. Is there a least number of generated images to make this model effective？

Maybe .  The metric of effective  is Whether improve the AD performence . In my paper , I  train a supervise model like DeSTseg based on generative dataset.

**Comment by TerryMelody** [comment]
> > > > > Dear authors! Thanks for your contributions! When running python generate_dataset.py, you set GENERATENUM = 20000 to generate target images of trainset. It seems a large number for only a defect type of a class. What's the influence of different image numbers of trainset, for example, if I set GENERATENUM = 200?
> > > > 
> > > > 
> > > > I also have this doubt, MvtecAD data set has a total of 15 classes, each class contains several small classes, each small class claims that 20000 images is too time-consuming and the storage space is large, if I only train one class can, have you tried it?
> > > 
> > > 
> > > Due to time constraints ， I only train 7 class of mvtec.
> > 
> > 
> > I got it. Thanks for your reply. Is there a least number of generated images to make this model effective？
> 
> Maybe . The metric of effective is Whether improve the AD performence . In my paper , I train a supervise model like DeSTseg based on generative dataset.

Thank you very much! A last little question: what's function of Foreground_predictor? It seems no use when generating trainset.

**Comment by TerryMelody** [comment]
you set trainer = pl.Trainer(gpus=1, precision=32, callbacks=[logger], max_epochs=10)
